routing/routing_helpers.cpp

```diff
@@ -18,6 +20,75 @@ namespace routing
 using namespace std;
 using namespace traffic;
 
+void FillSegmentInfo(vector<Segment> const & segments, vector<Junction> const & junctions,
+                     Route::TTurns const & turnDirs, Route::TStreets const & streets,
+                     Route::TTimes const & times, shared_ptr<TrafficStash> const & trafficStash,
+                     vector<Route::SegmentInfo> & segmentInfo)
+{
+  CHECK_EQUAL(segments.size() + 1, junctions.size(), ());
+  CHECK(!turnDirs.empty(), ());
+  CHECK(!times.empty(), ());
+
+  CHECK(std::is_sorted(times.cbegin(), times.cend(), my::LessBy(&Route::TTimeItem::first)), ());
+  CHECK(std::is_sorted(turnDirs.cbegin(), turnDirs.cend(), my::LessBy(&turns::TurnItem::m_index)), ());
+  CHECK(std::is_sorted(streets.cbegin(), streets.cend(), my::LessBy(&Route::TStreetItem::first)), ());
+
+  segmentInfo.clear();
+  if (segments.empty())
+    return;
+
+  segmentInfo.reserve(segments.size());
+  // Note. |turnDirs|, |streets| and |times| have point()|junctions| based index.
```

Comment:
I wanted to write "point based index". That means 0 is a start point of the route. 1 is the next end of start segment and so on. So the text should have been:
Note. |turnDirs|, |streets| and |times| have point(|junctions|) based index.